src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -1,31 +1,30 @@
 /*
  * File: z_en_sth.c
  * Overlay: ovl_En_Sth
- * Description: Guy looking at moon in South Clock Town
+ * Description: Guy looking at moon in South Clock Town,
+ *              And the Cured Swamp spiderhouse man who gives you Mask of Truth
+ *              And the man who wants to buy Ocean Spiderhouse from you with rupees or Wallet
  */
 
 #include "z_en_sth.h"
-#include "objects/object_sth/object_sth.h"
-#include "objects/object_ahg/object_ahg.h"
-#include "objects/object_mask_truth/object_mask_truth.h"
 
 #define FLAGS (ACTOR_FLAG_1 | ACTOR_FLAG_8)
 
 #define THIS ((EnSth*)thisx)
 
 void EnSth_Init(Actor* thisx, PlayState* play);
 void EnSth_Destroy(Actor* thisx, PlayState* play);
+void EnSth_WaitForObject(Actor* thisx, PlayState* play);
+
+void EnSth_PanicIdle(EnSth* this, PlayState* play);
+void EnSth_HandleOceanSpiderhouseConversation(EnSth* this, PlayState* play);
+void EnSth_OceanSpiderhouseIdle(EnSth* this, PlayState* play);
+void EnSth_MoonLookingIdle(EnSth* this, PlayState* play);
+void EnSth_DefaultIdle(EnSth* this, PlayState* play);
+void EnSth_HandleSwampSpiderhouseConversation(EnSth* this, PlayState* play);
+void EnSth_SwampSpiderhouseCuredIdle(EnSth* this, PlayState* play);
```

Comment:
Kind of realized something looking at this: specifying "Cured" here isn't necessary, right? Since the cursed version is a completely different actor. For me, I'd just remove it, since that would make the Swamp and Oceanside functions have similar names, but what do you think?